### PR TITLE
M73 Implementation

### PR DESCRIPTION
**Pull request by CampbellFabrications**
Added code to support #6451 
Editing may be needed for the # include methods but its current implementation works

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -7258,6 +7258,14 @@ inline void gcode_M42() {
 
 #endif // AUTO_BED_LEVELING_UBL && UBL_G26_MESH_VALIDATION
 
+/**M73: Set LCD Progress Bar Percentage */ 
+#if ENABLED(LCD_PROGRESS_BAR) 
+  inline void gcode_M73() { 
+  #include "ultralcd_impl_HD44780.h"
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:7264]: These includes are dangerous here, and should only ever be included at the top of `ultralcd.cpp`.

I like the `M73` concept, but this is not the correct implementation. Let me see if I can draw something up that shows the correct methodology.

Comment (CampbellFabrications) [reply Marlin/Marlin_main.cpp:7264]: Could you add the labels "Feature Request" and "Needs Work"?

**Comment by CampbellFabrications** [comment]
It is also a point of consideration regarding restrictions:
If someone without an LCD uses the command, will it return "command not found" or will it just do nothing. I'm inclined to think the latter (which will require additional coding), because M73 P100 should trigger the end timer, whereas M73 P0 should trigger the start timer

**Comment by thinkyhead** [comment]
See #7986 for the full implementation. It should make sense on first reading. Thanks for pushing us to get this included!

**Comment by thinkyhead** [comment]
> because M73 P100 should trigger the end timer, whereas M73 P0 should trigger the start timer

I'm inclined to simply regard `M73` as a cosmetic courtesy, to allow the progress bar to be set from the host side in the same manner that it can set a print job as being active or not. It shouldn't come with side-effects, nor act as a command to initiate other actions.

The standard behavior in Marlin is to ignore unknown G-codes and only throw errors on bad parameters for known G-codes.

**Comment by thinkyhead** [comment]
Perhaps what we're looking for is actually `M532`. The notifications used by MakerBot with `M73` are not important for non-MakerBot systems.

**Comment by dot-bob** [comment]
Repetier firmware and MK4duo uses M532 in this way to report progress.  It would make marlin one step closer to progress reporting using the firmware capabilities protocol: [http://www.reprap.org/wiki/Firmware_Capabilities_Protocol#Cap:PROGRESS](http://www.reprap.org/wiki/Firmware_Capabilities_Protocol#Cap:PROGRESS)
 see #5139

**Pull request closed without merging**